articles/migrations/guides/legacy-lock-api-deprecation.md

```diff
@@ -0,0 +1,195 @@
+
+::: note
+The `getSSOData()` and `checkSession()` functions should only be used from a Single Page Application
+:::
+
+##### checkSession
+
+* The Auth0.js `checkSession()` function only checks whether or not a user has an existing session in Auth0.
+* Invoking the `checkSession()` function will trigger an /authorize call, which will in turn result in the execution of [rules](/rules).
+* The new `checkSession()` function is more lightweight and should be used as a replacement for `getSSOData()` unless `getSSOData()` features are needed.
+
+##### getSSOData
+
+* The Auth0.js v9 `getSSOData()` function will continue to work, but it now [behaves differently than in the past](/libraries/auth0js/v9/migration-v8-v9#review-calls-to-getssodata-).
+* In Auth0.js v9, `getSSOData()` will check if a user has an existing session and perform a further check to determine if the user is the same one as in the last interactive authentication transaction. This supports Lock’s feature of showing the last logged-in user to facilitate subsequent logins.
+* Invoking the `getSSOData()` function will now trigger a call to the [/authorize](/api/authentication#authorize-client) endpoint, which will in turn result in the execution of [rules](/rules).
+
+##### Polling for an existing session
+
+In some multi-application scenarios, where a user logging out of one application needs to be logged out of other applications, an application may have been set up to periodically poll Auth0 using `getSSOData()` to see if a session existed, and if not, log the user out of the application. 
+
+Instead of doing this, applications should now use `checkSession()` instead of `getSSOData()`. The `getSSOData()` function performs more work behind the scenes than is needed for this purpose and applications that are not switched to `checkSession()` will suffer a needless performance penalty.
+
+The poll interval between checks to `checkSession()` should be at least 10 seconds between calls to avoid any issues in the future with rate limiting of this call.
+
+#### Web applications
+
+In "web applications", the backend typically has a session for the user. Over time, the application session may expire, in which case the application should renew the session. The application backend should invoke a call to the [/authorize](/api/authentication#authorize-client) endpoint to get a new token. If the Authorization Server (Auth0 in this case) still has a session for the user, the user will not have to re-enter their credentials to log in again. If Auth0 no longer has a session for the user, the user has to log in again.
+
+Customers with web applications who make server-side calls as part of the [Authorization Code Grant Flow](/api-auth/grant/authorization-code), and use the `response_type = code`, should use this approach. Specifically, such applications should make an /authorize request to refresh the token, which effectively renews the application session.
+
```

Comment:
I find this paragraph confusing, "Customers with web applications who make server-side calls as part of the Authorization Code Grant Flow", do we mean that they call an API from the backend? If so, I would say that they call an API from the backend, not that they do it as 'part of the auth code grant flow'.

And it's not that they call /authorize to refresh the token, they do this https://auth0.com/docs/tokens/refresh-token/current#use-a-refresh-token for refreshing a token, and that does not "renew the application session", it renews the token.